astroquery/cadc/core.py

```diff
@@ -270,80 +308,158 @@ def get_data_urls(self, query_result, include_auxiliaries=False):
         return result
 
-    def get_tables(self, only_names=False, verbose=False):
+    def get_tables(self, only_names=False, verbose=None):
         """
         Gets all public tables
 
         Parameters
         ----------
-        only_names : bool, TAP+ only, optional, default 'False'
+        only_names : bool, optional, default 'False'
             True to load table names only
-        verbose : bool, optional, default 'False'
-            flag to display information about the process
+        verbose : deprecated
 
         Returns
         -------
         A list of table objects
         """
-        return self._cadctap.load_tables(only_names, verbose)
+        if verbose is not None:
+            warnings.warn('verbose deprecated since 0.4.0')
+        table_set = self.cadctap.tables
+        if only_names:
+            return list(table_set.keys())
+        else:
+            return list(table_set.values())
 
-    def get_table(self, table, verbose=False):
+    def get_table(self, table, verbose=None):
         """
         Gets the specified table
 
         Parameters
         ----------
         table : str, mandatory
             full qualified table name (i.e. schema name + table name)
-        verbose : bool, optional, default 'False'
-            flag to display information about the process
+        verbose : deprecated
 
         Returns
         -------
         A table object
         """
-        return self._cadctap.load_table(table, verbose)
+        if verbose is not None:
+            warnings.warn('verbose deprecated since 0.4.0')
+        tables = self.get_tables()
+        for t in tables:
+            if table == t.name:
+                return t
+
+    def exec_sync(self, query, maxrec=None, uploads=None, output_file=None):
+        """
+        Run a query and return the results or save them in a output_file
+
+        Parameters
+        ----------
+        query : str, mandatory
+            SQL to execute
+        maxrec : int
+            the maximum records to return. defaults to the service default
+        uploads:
+            Temporary tables to upload and run with the queries
+        output_file: str or file handler:
+            File to save the results to
 
-    def query_async(self, query):
+        Returns
+        -------
+        Results of running the query in (for now) votable format
+
+        Notes
+        -----
+        Support for other output formats (tsv, csv) to be added as soon
+        as they are available in pyvo.
         """
-        Runs a query and returns results in Table format
+        response = self.cadctap.search(query, language='ADQL',
+                                       uploads=uploads)
+        result = response.to_table()
+        if output_file:
+            if isinstance(output_file, str):
+                with open(output_file, 'bw') as f:
+                    f.write(result)
+                    return
```

Comment:
is there any particular reason not to return the `result` when writing to file?  I'd be inclined to remove both of the blank `return` statements here and have `result` always be returned, and `output_file` being written will then just be a side effect.  I'm not sure that's better, though...